src/bosh_openstack_cpi/spec/unit/manual_network_spec.rb

```diff
@@ -67,8 +67,23 @@
       end
 
       context 'allowed_address_pair' do
+        context 'is configured to disable IP source check' do
+          let(:network_spec) { manual_network_spec(ip: '10.0.0.1', overwrites: { 'cloud_properties' => { "security_groups"=>["default"], "net_id"=>"net",'vrrp_port_check'=> false} }) }
+          let(:allowed_address_pairs) { '1.1.1.1/0' }
+
+          before(:each) do
+            subject.allowed_address_pairs = allowed_address_pairs
+          end
+
+          it 'configures to disable port check allowed_address_pair' do
+            subject.prepare(openstack, security_groups_to_be_used)
+
+            expect(ports).to have_received(:create).once.with(network_id: anything, fixed_ips: anything, security_groups: ['default-security-group-id'], allowed_address_pairs: [{ ip_address: allowed_address_pairs }])
```

Comment:
Actually, we should check that `ports` didn't received the `all` call here like: `expect(ports).to_not have_received(:all).with(fixed_ips: "ip_address=#{allowed_address_pairs}")`

Reply:
On my first attempt, I added this check, but since it is the default behavior to raise an error on unexpected call, I removed it afterwards. 
It's not an issue to add this check.

Reply:
but you can check that it hasn't been called with `...to_not have_received...`

Reply:
@o-orand would you be able to make the changes requested by @beyhan?

Reply:
I made the changes. It requires to add `allow(ports).to receive(:all)` in test
to avoid 
```
  1) Bosh::OpenStackCloud::ManualNetwork#prepare with 'use_nova_networking=false' allowed_address_pair is configured to disable IP source check configures to disable port check allowed_address_pair
     Failure/Error: expect(ports).to_not have_received(:all).with(fixed_ips: "ip_address=#{allowed_address_pairs}")
       #<Double "Fog::OpenStack::Network::Ports"> expected to have received all, but that object is not a spy or method has not been stubbed.

```